test/integration/test_integration_basics.py

```diff
@@ -654,6 +658,33 @@ def test_http_request_release_timing(http_test_server_fixture, qps_parameterizat
     assertCounterEqual(counters, "benchmark.http_2xx", (total_requests))
 
 
+def send_sigterm(p):
+  # Sleep for a while, under tsan the client needs a lot of time
+  # to start up.
+  time.sleep(10)
+  p.terminate()
+
+
+def test_cancellation(http_test_server_fixture):
+  """
+  That that we can use signals to cancel execution.
+  """
+  args = [
+      http_test_server_fixture.nighthawk_client_path, "--concurrency", "2",
+      http_test_server_fixture.getTestServerRootUri(), "--duration", "1000", "--output-format",
+      "json"
+  ]
+  client_process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  Thread(target=(lambda: send_sigterm(client_process))).start()
+  stdout, stderr = client_process.communicate()
+  client_process.wait()
+  output = stdout.decode('utf-8')
+  assert (client_process.returncode == 0)
+  parsed_json = json.loads(output)
+  counters = http_test_server_fixture.getNighthawkCounterMapFromJson(parsed_json)
+  assertCounterEqual(counters, "graceful_stop_requested", 2)
```

Comment:
Might also want to assert that the 2xx counters are >0, so we know tsan didn't slow it down so much that it only exercised the early exit in ProcessImpl::runInternal when cancelled_ is already true.

Reply:
Sure; done in 1fddbec526a464606577a569cf8a8a464d175025